### ISSUE TITLE
Consider dual licensing under Apache 2

### ISSUE THREAD

**Issue by pcwalton** [servo/webrender#2836]
I've seen some potential embedders have concerns around the license. See https://twitter.com/Brittain_Ben/status/1009855014792097792 for example.

What do folks think about dual licensing with a more common license? We'd lose file level copyleft, but I'm not sure that's a particularly meaningful boundary, especially in Rust where external nested modules are trivial to create.

**Comment by Gankro** [comment]
i'm fine with relicensing my contributions in this way

**Comment by gw3583** [comment]
I personally have no problem changing the license.

I'm not sure what the exact process is - do we need signoff from all contributors (there are currently 96 contributors according to the insights page, although many of those are a small, single patch).

Do we also need sign-off from legal in Mozilla?

**Comment by pcwalton** [comment]
I believe that all contributions made by Mozilla employees during their employment will automatically be relicensed once the decision is made. For others, signoff is required. @aturon may have more experience with the process here.

**Comment by larsbergstrom** [comment]
Yes, we'd need to have this conversation internally first along with our legal and policy teams. Feel free to kick this off with me internally and I'll drive it.

**Comment by kvark** [comment]
FWIW, we've recently finished the conversion in https://github.com/gfx-rs/gfx/issues/847 and it was a huge effort.

**Comment by Manishearth** [comment]
Fine with relicensing under the Rust license (MIT/Apache 2) or any combination of MPL and those licenses.

**Comment by atouchet** [comment]
Was a decision ever made about this?

**Comment by larsbergstrom** [comment]
Unfortunately, not at this time. With the [move](https://mozillagfx.wordpress.com/2019/01/10/webrender-newsletter-35/) of WebRender to be a more critical & integrated part of Firefox and its product infrastructure, it is unlikely we would move to relicense. Mozilla code license policies currently prefer MPL2 for core product code.

Certainly, if there is a key project or partner request, feel free to get in touch with me. Unfortunately, to date, all that we've gotten through official channels is, "we'd like to be able to take this code, fork it privately, and not contribute back patches publicly but intend to ship products based on it" which is not a super compelling argument and preventing that behavior is literally a key reason we license code under MPL2 in the first place.
